src/openfermion/resource_estimates/pbc/thc/utils/gvec_logic.py

```diff
@@ -0,0 +1,184 @@
+# coverage: ignore
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import itertools
+import numpy as np
+from pyscf.lib.numpy_helper import cartesian_prod
+
+
+def get_miller_indices(kmesh):
+    """Calculate the miller indices on a gamma centered non-1stBZ Monkorhst-Pack
+      mesh
+
+    Args:
+      kmesh: 1-D iteratble with the number of k-points in the x,y,z direction
+        [Nk_x, NK_y, NK_z] where NK_x/y/z are positive integers
```

Comment:
so wait: is this argument always length-3? if yes: can we make that more explicitly documented and checked? for example, you could just tuple-unpack it

```python
nx, ny, nz = kmesh
if nx < 1: raise TypeError("x...")
...
```

or even factor out a helper function

```python

def _kmesh_helper(n):
  if n < 1: raise TypeError()
  return np.arange(n, dtype=float)

int_scaled_kpts = cartesian_prod([
  _helper(nx), _helper(ny), _helper(nz)])

Reply:
Done, decided against the refactor.